packages/lib-classifier/src/store/SubjectStore/SubjectStore.js

```diff
@@ -322,6 +322,11 @@ const SubjectStore = types
     }
 
     function reset () {
+      /*
+      This line stops the classifier from crashing when changing workflows.
+      TODO: It's a safeReference, so why is it not being cleared automatically?
+      */
+      self.active = undefined
```

Comment:
I'm not sure why yet, or if it's this line specifically, but when testing this branch with Davy Notebooks locally if I:
- go to workflow A
- draw a line (start a classification)
- go back to the project home page
- go to workflow B => I get an error including `TypeError: Cannot read properties of undefined (reading 'latest')`

Changing [TasksConnector line 27](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/TasksConnector.js#L27) to `const latest = subject?.stepHistory?.latest` (adding optional chaining to stepHistory) appears to fix the issue.

I didn't see an issue related to it, but [Task](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/Task/Task.js#L27) maybe could use the optional chaining as well.

Reply:
That’s interesting. I’d have thought that change would stop you from classifying at all, because that one is used to get the current step and its annotations.